### PR TITLE
Make most of the fasm2frames.py tests pass again.

### LINKED ISSUE #492: Fix or delete utils/test_fasm2frames.py
The utils/test_fasm2frames.py currently fails with;
```
if [ -e env/bin/activate ]; then . env/bin/activate; fi; py.test --ignore third_party --ignore .git --ignore env --ignore build --ignore fuzzers --ignore minitests --ignore experiments --doctest-modules --junitxml=build/py_test_results.xml                                                                                                                                                                                         
========================================================================================================================================================================================================= test session starts ==========================================================================================================================================================================================================
platform linux -- Python 3.5.4rc1, pytest-4.1.0, py-1.7.0, pluggy-0.8.1
rootdir: /usr/local/google/home/tansell/github/SymbiFlow/prjxray, inifile:
collected 15 items                                                                                                                                                                                                                                                                                                                                                                                                                     
                                
docs/markdown_code_symlinks.py .                                                                                                                                                                                                                                                                                                                                                                                                 [  6%]
prjxray/lib.py .                                                                                                                                                                                                                                                                                                                                                                                                                 [ 13%]
tests/test_util.py .                                                                                                                                                                                                                                                                                                                                                                                                             [ 20%]
utils/test_fasm2frames.py FFFFFFFFFFF                                                                                                                                                                                                                                                                                                                                                                                            [ 93%]
utils/xjson.py .                                                                                                                                                                                                                                                                                                                                                                                                                 [100%]
                                                                                                                                                    
=============================================================================================================================================================================================================== FAILURES ===============================================================================================================================================================================================================
____________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_badkey _____________________________________________________________________________________________________________________________________________________________________________________________________
                         
self = <test_fasm2frames.TestStringMethods testMethod=test_badkey>
                                                                                        
    def test_badkey(self):
        '''Bad key should throw syntax error'''
        fin = StringIO("CLBLM_L_X10Y102.SLICEM_X0.SRUSEDMUX 2")                                                                                                                                                                                                                                                                                                                                                                         
        fout = StringIO()
        try:                                                          
>           fasm2frames.run(fin, fout)
E           TypeError: run() missing 1 required positional argument: 'f_out'
                                                               
utils/test_fasm2frames.py:118: TypeError
____________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_dupkey _____________________________________________________________________________________________________________________________________________________________________________________________________
                                                                        
self = <test_fasm2frames.TestStringMethods testMethod=test_dupkey>
                                       
        def test_dupkey(self):                                                                                                                                                                                                                                                                                                                                                                                                          
            '''Duplicate key should throw syntax error'''
            fin = StringIO(                                                 
                """\
    CLBLM_L_X10Y102.SLICEM_X0.SRUSEDMUX 0
    CLBLM_L_X10Y102.SLICEM_X0.SRUSEDMUX 1                                       
    """)                                                     
            fout = StringIO()
            try:                  
>               fasm2frames.run(fin, fout)                              
E               TypeError: run() missing 1 required positional argument: 'f_out'
                                       
utils/test_fasm2frames.py:132: TypeError                                                                                                                                                                                                                                                                                                                                                                                                
____________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_ff_int _____________________________________________________________________________________________________________________________________________________________________________________________________
                                                                      
self = <test_fasm2frames.TestStringMethods testMethod=test_ff_int>
                              
    def test_ff_int(self):                                                  
        self.bitread_frm_equals(                              
>           'test_data/ff_int.fasm', 'test_data/ff_int/design.bits')
                         
utils/test_fasm2frames.py:78:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
                                                                            
self = <test_fasm2frames.TestStringMethods testMethod=test_ff_int>, frm_fn = 'test_data/ff_int.fasm', bitread_fn = 'test_data/ff_int/design.bits'
                                        
    def bitread_frm_equals(self, frm_fn, bitread_fn):                                                                                                                                                                                                                                                                                                                                                                                   
        fout = StringIO()
>       fasm2frames.run(open(frm_fn, 'r'), fout)                  
E       FileNotFoundError: [Errno 2] No such file or directory: 'test_data/ff_int.fasm'
                          
utils/test_fasm2frames.py:60: FileNotFoundError          
___________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_ff_int_0s ___________________________________________________________________________________________________________________________________________________________________________________________________
    
self = <test_fasm2frames.TestStringMethods testMethod=test_ff_int_0s>
                                                                    
    def test_ff_int_0s(self):                                                           
        '''Explicit 0 entries'''
        self.bitread_frm_equals(                
>           'test_data/ff_int_0s.fasm', 'test_data/ff_int/design.bits')                                                                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                                                                                                                                        
utils/test_fasm2frames.py:111:                 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ [0/9154]

self = <test_fasm2frames.TestStringMethods testMethod=test_ff_int_0s>, frm_fn = 'test_data/ff_int_0s.fasm', bitread_fn = 'test_data/ff_int/design.bits'

    def bitread_frm_equals(self, frm_fn, bitread_fn):
        fout = StringIO()
>       fasm2frames.run(open(frm_fn, 'r'), fout)                            
E       FileNotFoundError: [Errno 2] No such file or directory: 'test_data/ff_int_0s.fasm'
                                                                   
utils/test_fasm2frames.py:60: FileNotFoundError
__________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_ff_int_op1 ___________________________________________________________________________________________________________________________________________________________________________________________________
                                                        
self = <test_fasm2frames.TestStringMethods testMethod=test_ff_int_op1>
                         
    def test_ff_int_op1(self):
        '''Omitted key set to '''              
        self.bitread_frm_equals(               
>           'test_data/ff_int_op1.fasm', 'test_data/ff_int/design.bits')

utils/test_fasm2frames.py:83:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
                                                          
self = <test_fasm2frames.TestStringMethods testMethod=test_ff_int_op1>, frm_fn = 'test_data/ff_int_op1.fasm', bitread_fn = 'test_data/ff_int/design.bits'
                                               
    def bitread_frm_equals(self, frm_fn, bitread_fn):
        fout = StringIO()                                          
>       fasm2frames.run(open(frm_fn, 'r'), fout)
E       FileNotFoundError: [Errno 2] No such file or directory: 'test_data/ff_int_op1.fasm'
                                                            
utils/test_fasm2frames.py:60: FileNotFoundError
______________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_lut ______________________________________________________________________________________________________________________________________________________________________________________________________

self = <test_fasm2frames.TestStringMethods testMethod=test_lut>
                                           
    def test_lut(self):               
        '''Simple smoke test on just the LUTs'''
        fout = StringIO()                                    
>       fasm2frames.run(open('test_data/lut.fasm', 'r'), fout)
E       FileNotFoundError: [Errno 2] No such file or directory: 'test_data/lut.fasm'
                                                                              
utils/test_fasm2frames.py:56: FileNotFoundError               
____________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_lut_int ____________________________________________________________________________________________________________________________________________________________________________________________________
                                                                                                                                                                                                                                                                                                                                                                                                                                        
self = <test_fasm2frames.TestStringMethods testMethod=test_lut_int>    
                                                                          
    def test_lut_int(self):                                                                                                                                                                                                                                                                                                                                                                                                            
        self.bitread_frm_equals(
>           'test_data/lut_int.fasm', 'test_data/lut_int/design.bits')                                                                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                                                                                                                                       
utils/test_fasm2frames.py:74:                                                                                                                                                                                                                                                                                                                                                                                                          
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
                                                                                                                                                                                                                                                                                                                                                                                                                                       
self = <test_fasm2frames.TestStringMethods testMethod=test_lut_int>, frm_fn = 'test_data/lut_int.fasm', bitread_fn = 'test_data/lut_int/design.bits'
                                                                                                                                                                                                                                                                                                                                                                                                                                        
    def bitread_frm_equals(self, frm_fn, bitread_fn):                                                                                                                                                                                                                                                                                                                                                                                   
        fout = StringIO()
>       fasm2frames.run(open(frm_fn, 'r'), fout)                  
E       FileNotFoundError: [Errno 2] No such file or directory: 'test_data/lut_int.fasm'
                          
utils/test_fasm2frames.py:60: FileNotFoundError
__________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_opkey_01_1 ___________________________________________________________________________________________________________________________________________________________________________________________________
                         
self = <test_fasm2frames.TestStringMethods testMethod=test_opkey_01_1>
                                      
    def test_opkey_01_1(self):                                              
        fin = StringIO("CLBLM_L_X10Y102.SLICEM_X0.SRUSEDMUX 1")
        fout = StringIO()               
>       fasm2frames.run(fin, fout)                                                                                                                                                                                                                                                                                                                                                                                                      
E       TypeError: run() missing 1 required positional argument: 'f_out'
                                                                  
utils/test_fasm2frames.py:95: TypeError
_______________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_opkey_01_default ________________________________________________________________________________________________________________________________________________________________________________________________
                                                         
self = <test_fasm2frames.TestStringMethods testMethod=test_opkey_01_default>
                    
    def test_opkey_01_default(self):     
        '''Optional key with binary omitted value should produce valid result'''
        fin = StringIO("CLBLM_L_X10Y102.SLICEM_X0.SRUSEDMUX")
        fout = StringIO()    
>       fasm2frames.run(fin, fout)
E       TypeError: run() missing 1 required positional argument: 'f_out'
                                                                                
utils/test_fasm2frames.py:90: TypeError
__________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_opkey_enum ___________________________________________________________________________________________________________________________________________________________________________________________________
                                                                                                                                                                                                                                                                                                                                                                                                                                        
self = <test_fasm2frames.TestStringMethods testMethod=test_opkey_enum>
                                                                  
    def test_opkey_enum(self):
        '''Optional key with enumerated value should produce syntax error'''
        # CLBLM_L.SLICEM_X0.AMUX.O6 !30_06 !30_07 !30_08 30_11
        fin = StringIO("CLBLM_L_X10Y102.SLICEM_X0.AMUX.O6")         
        fout = StringIO()
        try:                 
>           fasm2frames.run(fin, fout)                                                                                                                                                                                                                                                                                                                                                                                                 
E           TypeError: run() missing 1 required positional argument: 'f_out'
                                                                                                                                                 
utils/test_fasm2frames.py:103: TypeError
____________________________________________________________________________________________________________________________________________________________________________________________________ TestStringMethods.test_sparse _____________________________________________________________________________________________________________________________________________________________________________________________________
                         
self = <test_fasm2frames.TestStringMethods testMethod=test_sparse>
                                                                                       
    def test_sparse(self):
        '''Verify sparse equivilent to normal encoding'''
        frm_fn = 'test_data/lut_int.fasm'                                                                                                                                                                                                                                                                                                                                                                                               
    
        fout_sparse = StringIO()                                     
>       fasm2frames.run(open(frm_fn, 'r'), fout_sparse, sparse=True)
E       FileNotFoundError: [Errno 2] No such file or directory: 'test_data/lut_int.fasm'
                                
utils/test_fasm2frames.py:142: FileNotFoundError
---------------------------------------------------------------------------------------------------------------------------------------------------------------- generated xml file: /usr/local/google/home/tansell/github/SymbiFlow/prjxray/build/py_test_results.xml -----------------------------------------------------------------------------------------------------------------------------------------------------------------
================================================================================================================================================================================================= 11 failed, 4 passed in 0.72 seconds ==================================================================================================================================================================================================
Makefile:28: recipe for target 'test-py' failed
make: *** [test-py] Error 1  
```

### PR DESCRIPTION
**Pull request by mithro**
Fixes #492

Signed-off-by: Tim 'mithro' Ansell <me@mith.ro>

### CODE REVIEW

**Comment by mithro** [comment]
This doesn't work yet....

File: database/artix7/settings.sh [hunk 1]
```diff
@@ -1 +1,25 @@
-source $(dirname ${BASH_SOURCE[0]})/../../settings/artix7.sh
```
Comment (mcmasterg) [inline database/artix7/settings.sh]: Whats with the settings.sh on this PR?

Comment (mithro) [reply database/artix7/settings.sh]: I "totally" didn't _accidentally_ commit that :-P

Fixed.

**Comment by tmichalak** [comment]
Is there any reason for this PR not getting merged?

**Pull request merged**